x/ibc/02-client/types/proposal.go

```diff
@@ -0,0 +1,66 @@
+package types
+
+import (
+	fmt "fmt"
+	"strings"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+)
+
+const (
+	// ProposalTypeClientUpdate defines the type for a ClientUpdateProposal
+	ProposalTypeClientUpdate = "ClientUpdate"
+)
+
+// Assert ClientUpdateProposal implements govtypes.Content at compile-time
+var _ govtypes.Content = &ClientUpdateProposal{}
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeClientUpdate)
+	govtypes.RegisterProposalTypeCodec(&ClientUpdateProposal{}, "cosmos-sdk/ClientUpdateProposal")
+}
+
+// NewClientUpdateProposal creates a new client update proposal.
+func NewClientUpdateProposal(title, description, clientID string, header exported.Header) (*ClientUpdateProposal, error) {
+	headerBytes, err := header.MarshalBinaryBare()
+	if err != nil {
+		return &ClientUpdateProposal{}, err
+	}
+
+	return &ClientUpdateProposal{title, description, clientID, headerBytes}, nil
+}
+
+// GetTitle returns the title of a community pool spend proposal.
+func (cup *ClientUpdateProposal) GetTitle() string { return cup.Title }
+
+// GetDescription returns the description of a community pool spend proposal.
+func (cup *ClientUpdateProposal) GetDescription() string { return cup.Description }
+
+// GetDescription returns the routing key of a community pool spend proposal.
+func (cup *ClientUpdateProposal) ProposalRoute() string { return RouterKey }
+
+// ProposalType returns the type of a community pool spend proposal.
+func (cup *ClientUpdateProposal) ProposalType() string { return ProposalTypeClientUpdate }
+
+// ValidateBasic runs basic stateless validity checks
+func (cup *ClientUpdateProposal) ValidateBasic() error {
+	err := govtypes.ValidateAbstract(cup)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// String implements the Stringer interface.
+func (cup ClientUpdateProposal) String() string {
```

Comment:
no need for this function, it should already be provided by proto if you remove the gogoproto.stringer option